### PR TITLE
Detect mismatch between requested and used cores (HPC-5468)

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/jobs/pbs/jobs.py [hunk 1]
```diff
@@ -172,6 +172,14 @@ def get_userjob_stats():
             ustat[-1].append(name)
             continue
 
+        if 'exec_hosts' in derived:
+            used_cores = sum(derived['exec_hosts'].values())
+            # derived['cores'] exists here (see nodes test above)
+            if not derived['nodes']*derived['cores'] == used_cores:
+                faults.append(('Mismatch requested/running cores in job %s. Marked as other.' % (name), jobdata))
```
Comment (boegel) [inline lib/vsc/jobs/pbs/jobs.py]: mention derived + used counts in message?

what does the 'marked as other' thing mean?

Comment (boegel) [inline lib/vsc/jobs/pbs/jobs.py]: also, no use for `(name)`, just use `name`

File: lib/vsc/jobs/pbs/jobs.py [hunk 2]
```diff
@@ -172,6 +172,14 @@ def get_userjob_stats():
...
+                ustat[-1].append(name)
+                continue
+
         nodes = derived['nodes']
         cores = derived['cores']
         corenodes = nodes * cores
```
Comment (boegel) [inline lib/vsc/jobs/pbs/jobs.py]: move this up, since you're using this in the check above?

**Comment by stdweird** [comment]
@boegel remarks addressed

File: lib/vsc/jobs/pbs/jobs.py [hunk 3]
```diff
@@ -176,6 +176,13 @@ def get_userjob_stats():
...
 
+        if 'exec_hosts' in derived:
+            used_cores = sum(derived['exec_hosts'].values())
+            if not corenodes == used_cores:
+                faults.append(('Mismatch requested %s /running %s cores in job %s. Marked as other.' % (corenodes, used_cores, name), jobdata))
```
Comment (boegel) [inline lib/vsc/jobs/pbs/jobs.py]: line (way) too long, please fix so this can be merged

**Comment by stdweird** [comment]
@boegel done

**Comment by boegel** [comment]
quoting @stdweird: "jenkins unittests are part of the submitfilter PR"

**Pull request merged**